File: android/src/main/java/com/reactnativehypertracksdk/common/Geotag.kt
package com.reactnativehypertracksdk.common

/**
 * The data that represents geotag to create. Now it contains only payload data but
 * can have other params like expectedLocation in future
 *
 * @param data Geotag payload
 */
internal data class Geotag(val data: Map<String, Any?>)

File: ios/common/Geotag.swift
import HyperTrack

struct Geotag {
    val data: Dictionary<String, Any>
    val expectedLocation: HyperTrack.Location,

    constructor(data: Dictionary<String, Any>, expectedLocation: HyperTrack.Location) {
        this.data = data
        this.expectedLocation = expectedLocation
    }
}